common/src/main/java/com/microsoft/identity/common/internal/request/SdkType.java

```diff
@@ -22,10 +22,24 @@
 // THE SOFTWARE.
 package com.microsoft.identity.common.internal.request;
 
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+
 /**
  * Enum to indicate if the request came from an app hosting Adal or Msal Sdk
  */
 public enum SdkType {
     ADAL,
-    MSAL
-}
\ No newline at end of file
+    MSAL,
+    MSALCPP;
+
+
+
+    public String getProductName() {
+        if ((SdkType.ADAL == this) || (SdkType.MSAL == this)) {
+            return AuthenticationConstants.SdkPlatformFields.PRODUCT_NAME_MSAL;
+        } else {
+            return AuthenticationConstants.SdkPlatformFields.PRODUCT_NAME_MSAL_CPP;
+        }
+
+    }
```

Comment:
Same comment mentioned applies here as well, since we are hardcoding the values in every request, there isnt a way where we could encounter any funny values, and hence only the values used in the enum would be considered for getProductName.

Reply:
This enumeration looks like it exists on the interface between broker and MSAL - is that correct?

Reply:
This means that at some point this exists as a series of bytes in IPC.  You do not know what is going to land on the other side, since it can be newer or older than what you expect.  We're going to decouple this layer in the broker to allow for us to pivot the serialization mechanisms more smoothly, but this means that there will be cases where you will not be able to put the SDKType into a bucket with which you are familiar. What will you do in that case?

Reply:
@AdamBJohnsonx In this case, this wont be a blocker since, for the mentioned cases : 

1. **new MSAL - old broker** : the old broker wont have the sdktype and even if the new MSAL sends the new sdktype (msal.cpp) the broker will neglect it and use the hardcoded value of sdktype (msal)

2. **old msal - new broker** : since old msal wont be sending the sdktype, the new broker has a default value of selecting msal as sdktype.

3. **old msal - old broker** : working fine

4. **new msal - new broker** : working fine

I am going to be testing these e2e scenarios just to make sure, we see the flow we are expecting to see, I would post the results here.

Reply:
Why is it OK if the old broker cannot accurately report the sdk type?

Reply:
Since the old broker is already deployed and is in production(the current implementation) is using the common which uses the hardcoded value? I mean to say, this added new functionality wont break the functionality of the broker versions already in the market today not having this change.

Reply:
Let's just make sure that having the divergence won't crash the broker in the event of an either-way-mismatch, and you're good to go after that.  We'll take on making this forwards-compatible for the next SDK addition later, and revisit any parameter that's similarly dangerous.